File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int n = nums.length;
        boolean[] demo =  new boolean[n+1];
        for(int num  : nums)
        {
            if(num>0 && num <=n)
            {
                demo[num]=true;
            }
        }
        for(int i = 1;i<=n;i++)
        {
            if(!demo[i])
            return i;
        }
        return n+1;
    }
}

File: 0041-first-missing-positive/0041-first-missing-positive.c
int firstMissingPositive(int* nums, int numsSize) {
    int n = numsSize;
    bool* demo = calloc(n+1,sizeof(bool));
    for(int i=0 ; i<n; i++)
    {
        int num = nums[i];
        if(num>0 && num<=n)
        {
            demo[num]=true;
        }
    }
    for(int i=1;i<=n;i++)
    {
        if(!demo[i])
        return i;
    }
    return n+1;
}